features/FEATURE_LWIP/lwip-interface/ppp_lwip.cpp

```diff
@@ -0,0 +1,322 @@
+#endif /* PPP_IPV6_SUPPORT */
+            break;
+        }
+        case PPPERR_PARAM: {
+            printf("status_cb: Invalid parameter\n");
+            break;
+        }
+        case PPPERR_OPEN: {
+            printf("status_cb: Unable to open PPP session\n");
+            break;
+        }
+        case PPPERR_DEVICE: {
+            printf("status_cb: Invalid I/O device for PPP\n");
+            break;
+        }
+        case PPPERR_ALLOC: {
+            printf("status_cb: Unable to allocate resources\n");
+            break;
+        }
+        case PPPERR_USER: {
+            printf("status_cb: User interrupt\n");
+            break;
+        }
+        case PPPERR_CONNECT: {
+            printf("status_cb: Connection lost\n");
+            break;
+        }
+        case PPPERR_AUTHFAIL: {
+            printf("status_cb: Failed authentication challenge\n");
+            break;
+        }
+        case PPPERR_PROTOCOL: {
+            printf("status_cb: Failed to meet protocol\n");
+            break;
+        }
+        case PPPERR_PEERDEAD: {
+            printf("status_cb: Connection timeout\n");
+            break;
+        }
+        case PPPERR_IDLETIMEOUT: {
+            printf("status_cb: Idle Timeout\n");
+            break;
+        }
+        case PPPERR_CONNECTTIME: {
+            printf("status_cb: Max connect time reached\n");
+            break;
+        }
+        case PPPERR_LOOPBACK: {
+            printf("status_cb: Loopback detected\n");
+            break;
+        }
+        default: {
+            printf("status_cb: Unknown error code %d\n", err_code);
+            break;
+        }
+    }
+
+#if 0
+    /*
+     * This should be in the switch case, this is put outside of the switch
+     * case for example readability.
+     */
+
+    if (err_code == PPPERR_NONE) {
+        return;
+    }
+
+    /* ppp_close() was previously called, don't reconnect */
+    if (err_code == PPPERR_USER) {
+        /* ppp_free(); -- can be called here */
+        return;
+    }
+
+    /*
+     * Try to reconnect in 30 seconds, if you need a modem chatscript you have
+     * to do a much better signaling here ;-)
+     */
+    ppp_connect(pcb, 30);
+    /* OR ppp_listen(pcb); */
+
+#endif
+}
+
+static void flush(FileHandle *stream)
+{
+    char buffer[8];
+    for (;;) {
+        ssize_t ret = stream->read(buffer, sizeof buffer);
+        if (ret <= 0) {
+            break;
+        }
+    }
+}
+
+#if !PPP_INPROC_IRQ_SAFE
+#error "PPP_INPROC_IRQ_SAFE must be enabled"
+#endif
+static void ppp_input(FileHandle *stream)
+{
+    // Allow new events from now, avoiding potential races around the read
+    event_queued = false;
+
+    if (!my_stream) {
+        return;
+    }
+
+    // Non-blocking error check handler
+    pollfh fhs;
+    fhs.fh = my_stream;
+    fhs.events = POLLIN;
+    poll(&fhs, 1, 0);
+    if (fhs.revents & (POLLHUP|POLLERR|POLLNVAL)) {
+        goto hup;
```

Comment:
Is there better way for this goto ? hup is just one condition, used 2 here. small helper function? Or?